scipy/sparse/linalg/tests/test_interface.py

```diff
@@ -63,22 +64,40 @@ def get_matvecs(A):
             assert_equal(A.dot(matrix([[1],[2],[3]])), [[14],[32]])
 
             assert_equal((2*A)*[1,1,1], [12,30])
-            assert_equal((2*A).rmatvec([1,1]), [10, 14, 18])
+            assert_equal((2 * A).rmatvec([1, 1]), [10, 14, 18])
             assert_equal((2*A).H.matvec([1,1]), [10, 14, 18])
             assert_equal((2*A)*[[1],[1],[1]], [[12],[30]])
-            assert_equal((2*A).matmat([[1],[1],[1]]), [[12],[30]])
+            assert_equal((2 * A).matmat([[1], [1], [1]]), [[12], [30]])
             assert_equal((A*2)*[1,1,1], [12,30])
             assert_equal((A*2)*[[1],[1],[1]], [[12],[30]])
             assert_equal((2j*A)*[1,1,1], [12j,30j])
             assert_equal((A+A)*[1,1,1], [12, 30])
-            assert_equal((A+A).rmatvec([1,1]), [10, 14, 18])
+            assert_equal((A + A).rmatvec([1, 1]), [10, 14, 18])
             assert_equal((A+A).H.matvec([1,1]), [10, 14, 18])
             assert_equal((A+A)*[[1],[1],[1]], [[12], [30]])
             assert_equal((A+A).matmat([[1],[1],[1]]), [[12], [30]])
             assert_equal((-A)*[1,1,1], [-6,-15])
             assert_equal((-A)*[[1],[1],[1]], [[-6],[-15]])
             assert_equal((A-A)*[1,1,1], [0,0])
-            assert_equal((A-A)*[[1],[1],[1]], [[0],[0]])
+            assert_equal((A - A) * [[1], [1], [1]], [[0], [0]])
+
+            X = np.array([[1, 2], [3, 4]])
```

Comment:
Probably better to move these into their own method. Also would be interesting if there were a way to write them so that you could use `pytest.mark.parametrize` as currently they are a little hard to read. (That could also just be a formatting issue though.)

Reply:
Sorry, my python knowledge is too low even to understand your suggestion...

I have just followed the style of other examples here, except that the left expression asserted is an actual formula, rather then the pre-computed result. That makes the lines longer, but surely easier to follow, since one is not forced to calculate the outcome manually...

Please advise.

Reply:
Sorry, I do not know how to do it. Any advice?

Reply:
This set of tests is sort of messy with all the hardcoded results, quite old legacy code...
I'd postpone cleaning this up to a separate PR.